### PR TITLE
hal: esp32c6: Power management support

### PR DESCRIPTION
**Pull request by raffarost**
Add LL files for power management for C6

### CODE REVIEW

**Review by sylvioalves — APPROVED** [review]

**Comment by sylvioalves** [comment]
@marekmatej @LucasTambor PTAL

File: components/esp_hw_support/sleep_modes.c [hunk 1]
```diff
@@ -406,11 +403,11 @@ void esp_deep_sleep_deregister_hook(esp_deep_sleep_cb_t old_dslp_cb)
 // [refactor-todo] provide target logic for body of uart functions below
 static void IRAM_ATTR flush_uarts(void)
 {
-    for (int i = 0; i < SOC_UART_NUM; ++i) {
+    for (int i = 0; i < SOC_UART_HP_NUM; ++i) {
```
Comment (marekmatej) [inline components/esp_hw_support/sleep_modes.c:406]: Does it mean that LP uarts are never suspended?

Comment (raffarost) [reply components/esp_hw_support/sleep_modes.c:406]: I believe so. This is following current implementation for IDF

File: components/esp_hw_support/sleep_clock.c
Comment (LucasTambor) [inline components/esp_hw_support/sleep_clock.c:1]: Is there a problem with the header order?

**Review by LucasTambor — APPROVED** [review]

File: components/esp_hw_support/sleep_clock.c
Comment (raffarost) [reply components/esp_hw_support/sleep_clock.c:1]: This is to fix a duplicate definition of the symbol ARRAY_SIZE.
Changing the order fixes the error as `esp_regdma.h` does the check before defining the symbol again.

**Pull request merged**